src/app/cli/src/coda_restarts_and_txns_holy_grail.ml

```diff
@@ -0,0 +1,74 @@
+open Core
+open Async
+open Coda_worker
+open Coda_main
+open Coda_base
+
+let name = "coda-restarts-and-txns-holy-grail"
+
+let main n () =
+  assert (n > 1) ;
+  let log = Logger.create () in
+  let log = Logger.child log name in
+  let snark_work_public_keys = function
+    | 0 ->
+        Some
+          (List.nth_exn Genesis_ledger.accounts 5 |> snd |> Account.public_key)
+    | _ -> None
+  in
+  let%bind testnet =
+    Coda_worker_testnet.test log n Option.some snark_work_public_keys
+      Protocols.Coda_pow.Work_selection.Seq
+  in
+  (* SEND TXNS *)
+  let sender_sk =
+    Genesis_ledger.largest_account_exn () |> fst |> Option.value_exn
+  in
+  let receiver_pk =
+    Genesis_ledger.largest_account_exn () |> snd |> Account.public_key
+  in
+  don't_wait_for
+  @@ Coda_worker_testnet.Payments.send_several_payments testnet ~node:0
+       ~src:sender_sk ~dest:receiver_pk ;
+  (* RESTART NODES *)
+  (* catchup *)
+  let idx = Quickcheck.random_value (Int.gen_incl 1 (n - 1)) in
+  let%bind () =
+    Coda_worker_testnet.Restarts.trigger_catchup testnet
+      ~largest_account_keypair:(Genesis_ledger.largest_account_keypair_exn ())
+      ~log ~node:idx
+      ~payment_receiver:(n - 1 - idx)
+  in
+  (* bootstrap *)
+  let idx = Quickcheck.random_value (Int.gen_incl 1 (n - 1)) in
+  let%bind () =
+    Coda_worker_testnet.Restarts.trigger_bootstrap testnet
+      ~largest_account_keypair:(Genesis_ledger.largest_account_keypair_exn ())
+      ~log ~node:idx
+      ~payment_receiver:(n - 1 - idx)
+  in
+  (* random *)
+  let idx = Quickcheck.random_value (Int.gen_incl 1 (n - 1)) in
+  let duration =
+    Quickcheck.(
+      random_value
+        Generator.(Float.gen_incl 1. 5. >>| fun x -> Time.Span.of_min x))
+  in
+  let%bind () =
```

Comment:
what's this bit for?

Reply:
i.e., what makes it different from the above

Reply:
It turns off a node for a random duration instead of a duration that is precisely tuned to hit specific branches of code (just in case this hits other edge cases).